pkg/maps/nat/nat.go

```diff
@@ -115,24 +109,51 @@ func (m *Map) DumpReliablyWithCallback(cb bpf.DumpCallback, stats *bpf.DumpStats
 	return (&m.Map).DumpReliablyWithCallback(cb, stats)
 }
 
-// DoDumpEntries iterates through Map m and writes the values of the
-// nat entries in m to a string.
-func DoDumpEntries(m NatMap) (string, error) {
+// DumpEntriesWithTimeDiff iterates through Map m and writes the values of the
+// nat entries in m to a string. If clockSource is not nil, it uses it to
+// compute the time difference of each entry from now and prints that too.
+func DumpEntriesWithTimeDiff(m NatMap, clockSource *models.ClockSource) (string, error) {
+	var toDeltaSecs func(uint64) string
 	var sb strings.Builder
 
-	nsecStart, _ := bpf.GetMtime()
+	if clockSource == nil {
+		toDeltaSecs = func(t uint64) string {
+			return fmt.Sprintf("? (raw %d)", t)
+		}
+	} else {
+		now, err := timestamp.GetCTCurTime(clockSource)
+		if err != nil {
+			return "", err
+		}
+		tsecNow, err := timestamp.ConvertCTTimeToSec(now, clockSource)
+		if err != nil {
+			return "", err
+		}
+		toDeltaSecs = func(t uint64) string {
+			tsec, _ := timestamp.ConvertCTTimeToSec(t, clockSource)
```

Comment:
We are checking for errs right above with a presumably valid clocksource and a known timestamp. 
That func can take adhoc timestamp so perhaps log the error since we can't return it?